Core/src/org/sleuthkit/autopsy/geolocation/KdTree.java

```diff
@@ -507,173 +373,258 @@ public EuclideanComparator(XYZPoint point) {
-            if (axis == Y_AXIS)
-                return Y_COMPARATOR.compare(o1, o2);
-            return Z_COMPARATOR.compare(o1, o2);
+        /**
+         * Returns the XYZ point of this node which contains the longitude and
+         * latitude values.
+         *
+         * @return XYZPoint
+         */
+        XYZPoint getPoint() {
+            return point;
         }
 
         /**
          * {@inheritDoc}
          */
         @Override
         public int hashCode() {
-            return 31 * (this.k + this.depth + this.id.hashCode());
+            return 31 * (DIMENSIONS + this.depth + this.getPoint().hashCode());
         }
 
         /**
          * {@inheritDoc}
          */
         @Override
         public boolean equals(Object obj) {
-            if (obj == null)
+            if (obj == null) {
                 return false;
-            if (!(obj instanceof KdNode))
+            }
+            if (!(obj instanceof KdNode)) {
                 return false;
+            }
 
             KdNode kdNode = (KdNode) obj;
-            if (this.compareTo(kdNode) == 0)
-                return true;
-            return false;
+            return (this.compareTo(kdNode) == 0);
         }
 
         /**
          * {@inheritDoc}
          */
         @Override
         public int compareTo(KdNode o) {
-            return compareTo(depth, k, this.id, o.id);
+            return compareTo(depth, this.getPoint(), o.getPoint());
         }
 
         /**
          * {@inheritDoc}
          */
         @Override
         public String toString() {
-            StringBuilder builder = new StringBuilder();
-            builder.append("k=").append(k);
-            builder.append(" depth=").append(depth);
-            builder.append(" id=").append(id.toString());
+            StringBuilder builder = new StringBuilder(200);
+            builder.append("dimensions=").append(DIMENSIONS).append(" depth=").append(depth).append(" point=").append(getPoint().toString());
             return builder.toString();
         }
     }
 
+    /**
+     * An XYZPoint is a representation of a three dimensional point.
+     *
+     * Z value will always been 0 when using latitude and longitude values.
+     */
     public static class XYZPoint implements Comparable<XYZPoint> {
 
         protected final double x;
         protected final double y;
         protected final double z;
 
         /**
-         * z is defaulted to zero.
+         * Constructs a new XYZPoint.
          *
-         * @param x
-         * @param y
+         * @param latitude
+         * @param longitude
          */
-        public XYZPoint(double x, double y) {
-            this.x = x;
-            this.y = y;
-            this.z = 0;
+        public XYZPoint(Double latitude, Double longitude) {
+            x = latitude;
+            y = longitude;
+            z = 0;
         }
 
         /**
-         * Default constructor
+         * Returns the x\latitude value for the point.
```

Comment:
Is that a backslash for a reason? Should it simply be x/latitude? makes it hard to read.